Sources/LLVM/ORCJIT.swift

```diff
@@ -0,0 +1,247 @@
+#if !NO_SWIFTPM
+import cllvm
+#endif
+
+/// A SymbolResolver is an object that has special logic for looking up symbols
+/// in the JIT.
+public protocol SymbolResolver {
+  /// Determines the callable address of the provided function in the JIT's
+  /// address space.
+  ///
+  /// - Parameters:
+  ///   - symbol: The name of the symbol to look up.
+  ///   - jit: The JIT in which the resolver is looking.
+  /// - Returns: `nil` if the symbol could not be found, otherwise the opaque
+  ///            address of a C function pointer that can be called to execute
+  ///            the compiled code.
+  func address(of symbol: String, in jit: ORCJIT) -> UInt64?
```

Comment:
Can we provide a public typealias for `LLVMOrcTargetAddress` instead of using UInt64?